### DISCUSSION TITLE
call $setTag to set string value but can only show part of the string content

### DISCUSSION THREAD

**Discussion by guolisongIsesol** [frangoteam/FUXA#924] [Q&A]
Hi FUXA team, i want to record the exact alarm time and show it in an output control or table grid. So i write a script and first set the alarm time into a string variable and then call $setTag to set the alarm time to the output control or table grid. But the problem is that, the alarm time content like "2023-10-7 12:34:23:560", but the control or table grid can only show "2023".  Can any one tell me how to solve this? Thanks in advance.

**Comment by guolisongIsesol** [comment]
I think i possiable find the cause. I read the source code, in runtime/devices/fuxaserver/index.js,  in the _parseValue function, i judge if the value is string, like this, 
`    var _parseValue = function (value) {
        if (typeof(value)=="string"){
            let val = value;
            return val;
        }else{
            let val = parseFloat(value);
            if (Number.isNaN(val)) {
                // maybe boolean
                val = Number(value);
                // maybe string
                if (Number.isNaN(val)) {
                    val = value;
                }
            } else {
                val = parseFloat(val.toFixed(5));
            }
            return val;
        }
    }`
Then the  total string can be shown in the control

**Comment by unocelli** [comment]
Hi, sorry for the delay.
Yes there is a problem to check if a string is a number, a Date is interpreted as a number. We will fix it.
